app/src/main/java/com/github/raulccabreu/redukt/ui/StateView.kt

```diff
@@ -0,0 +1,49 @@
+package com.github.raulccabreu.redukt.ui
+
+import android.content.Context
+import android.view.View
+import com.github.raulccabreu.redukt.Redukt
+import com.github.raulccabreu.redukt.states.StateListener
+
+abstract class StateView<T>(context: Context) : View(context), StateListener<T> {
+
+    private var isRegistered = false
+
+    protected abstract fun changeStateWhenHidden(): Boolean
+    protected abstract fun getRedukt(): Redukt<T>
+
+    override fun onAttachedToWindow() {
+        super.onAttachedToWindow()
+
+        registerReducer()
+    }
+
+    override fun onDetachedFromWindow() {
+        unregisterReducer()
+        super.onDetachedFromWindow()
+    }
+
+    override fun setVisibility(visibility: Int) {
```

Comment:
I don't think 100% sure that is necessary a hide/show register and unregister mechanism at views, if we will add this, maybe at activity and fragment its necessary to implement this behavior too at resume/show with the `changeStateWhenHidden` method (suggest to change the method name to: `listenChangesWhenHidden`).